service/lib/dinstaller/storage/commit_actions.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "singleton"
+require "y2storage/storage_manager"
+
+module DInstaller
+  module Storage
+    # Backend class to calculate a storage proposal
+    class CommitActions
+      include Singleton
+
+      def all_actions
+        main_actions + subvolume_actions
+      end
+
+      def main_actions
+        actions = actions.reject { |a| a.device_is?(:btrfs_subvolume) }
+        sort_actions(actions)
+      end
+
+      def subvolume_actions
+        actions = actions.select { |a| a.device_is?(:btrfs_subvolume) }
+        sort_actions(actions)
+      end
```

Comment:
regarding that signal. Can we get some observer that recognize if actions changes?

Reply:
Yes, I will implement some observer/callbacks system.

Reply:
After discussing it, we decided to not implement observers yet. Let's discuss first how to deal with threads.